logprep/framework/rule_tree/rule_parser.py

```diff
@@ -126,15 +130,36 @@ def _add_exists_filter(parsed_rules: list):
         for parsed_rule in parsed_rules:
             temp_parsed_rule = parsed_rule.copy()
             skipped_counter = 0
-
-            for segment_index, segment in enumerate(temp_parsed_rule):
-                if isinstance(segment, (Exists, Not, Always)):
+            for segment_idx, segment in enumerate(temp_parsed_rule):
+                if isinstance(segment, (Exists, Always)):
                     skipped_counter += 1
                     continue
 
-                exists_filter = Exists(segment.key)
-                if exists_filter in parsed_rule:
+                exists_filter = RuleParser._get_exists_filter(segment)
+                if exists_filter is None or exists_filter in parsed_rule:
                     skipped_counter += 1
-                    continue
+                else:
+                    parsed_rule.insert(segment_idx * 2 - skipped_counter, exists_filter)
+
+    @staticmethod
+    def _get_exists_filter(segment: FilterExpression) -> Optional[Exists]:
+        """Get Exists filter expression based in segment.
 
-                parsed_rule.insert(segment_index * 2 - skipped_counter, exists_filter)
+        Parameters
+        ----------
+        segment: FilterExpression
+            Filter expressions segment.
+
+        Returns
+        -------
+        Optional[Exists]
+            Returns an Exists FilterExpression from a KeyValueBased FilterExpression or None.
+
+        """
+        if isinstance(segment, Not):
+            not_child = segment.children[0]
+            if isinstance(not_child, KeyValueBasedFilterExpression):
+                return Exists(not_child.key)
+        elif isinstance(segment, KeyValueBasedFilterExpression):
+            return Exists(segment.key)
+        return None
```

Comment:
How about we use this implementation? It's a bit easier to read and understand by preventing nested if-statements.
```suggestion
        if isinstance(segment, Not):
            segment = segment.children[0]
        if isinstance(segment, KeyValueBasedFilterExpression):
            return Exists(segment.key)
        return None
```

Reply:
Thanks for the suggestion. I've changed it.